src/test/java/spoon/reflect/factory/PackageFactoryTest.java

```diff
@@ -29,4 +31,21 @@ void getOrCreate_returnsNestedPackageStructure_whenQualifiedNameRepeatsSimpleNam
 		assertThat(topLevelPackage.getPackage(topLevelPackageName), sameInstance(packageWithDuplicatedSimpleNames));
 		assertThat(packageWithDuplicatedSimpleNames.getParent(), sameInstance(topLevelPackage));
 	}
+	
+		@Test
+		@GitHubIssue(issueNumber = 5140, fixed = true)
+    void testGetPackageWithNameContainingDollarSign() {
+			// contract: A package with a name containing a dollar sign can be retrieved using the PackageFactory
```

Comment:
The formatting of this whole file is a bit broken

Reply:
There was some random mix between spaces and tabs. It should be fixed now.